prometheus/prometheus.go

```diff
@@ -39,19 +60,44 @@ type PrometheusSink struct {
 	expiration time.Duration
 }
 
-type PrometheusGauge struct {
+// GaugeDefinition can be provided to PrometheusOpts to declare a constant gauge that is not deleted on expiry.
+type GaugeDefinition struct {
+	Name []string
+	ConstLabels []metrics.Label
+	Help string
+}
+
+type gauge struct {
 	prometheus.Gauge
 	updatedAt time.Time
+	// canDelete is set if the metric is created during runtime so we know it's ephemeral and can delete it on expiry.
+	canDelete bool
+}
+
+// SummaryDefinition can be provided to PrometheusOpts to declare a constant summary that is not deleted on expiry.
+type SummaryDefinition struct {
+	Name []string
+	ConstLabels []metrics.Label
+	Help string
 }
 
-type PrometheusSummary struct {
+type summary struct {
```

Comment:
these classes were exported before (even though they didn't need to be), but i don't find any indication that they're used anywhere, so 👍

Reply:
Yeah we looked over it and it seemed pretty safe to unexport these because they're only used internally.